### PR TITLE
libhfs3 updates (xattr and patches)

### PR DESCRIPTION
**Pull request by sapanshah**
Reviwers: @immuta/developers 

These are all the C++ changes to libhdfs3 to:
* use libhdfs3 outside of the hawq project
* add the ability to list all the extended attributes for a path
* add an example of how to query the namenode for information

What still remains to be done before submitting this back as a patch:
* add setxattr, removexattr, getxattr
* add all the xattr stuff to the c library

### CODE REVIEW

File: depends/libhdfs3/osx/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Packging LibHdfs3 for Mac OS X
+This directory contains 
+- build script for osx for developign libhdfs3
```
Comment (bjstabi) [inline depends/libhdfs3/osx/README.md]: typo: developign = developing

**Review by bjstabi — APPROVED** [review]

File: depends/libhdfs3/src/server/NamenodeImpl.cpp [hunk 2]
```diff
@@ -865,5 +867,33 @@ bool NamenodeImpl::listEncryptionZones(const int64_t id, std::vector<EncryptionZ
     }
 }
 
+//Idempotent
+std::map<std::string, std::string> NamenodeImpl::listXAttrs(const std::string & src) {
+    try {
```
Comment (KyleLilly) [inline depends/libhdfs3/src/server/NamenodeImpl.cpp]: Strictly from a style standpoint - add comment with throws here?

File: depends/libhdfs3/src/server/NamenodeImpl.cpp [hunk 3]
```diff
@@ -865,5 +867,33 @@ bool NamenodeImpl::listEncryptionZones(const int64_t id, std::vector<EncryptionZ
...
+        ListXAttrsRequestProto request;
+        ListXAttrsResponseProto response;
+        request.set_src(src);
+        invoke(RpcCall(true, "listXAttrs", &request, &response));
+        
+        std::map<std::string, std::string> retval;
+        for(int i=0; i < response.xattrs_size(); ++i) {
```
Comment (KyleLilly) [inline depends/libhdfs3/src/server/NamenodeImpl.cpp]: Spacing - `i = 0`

**Review by KyleLilly — DISMISSED** [review]

File: depends/libhdfs3/osx/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+- homebrew formula to install libhdfs3 on osx
+
+## Installing libhdfs3
+All you need to do is run
+
+```bash
+./install_homebrew_pkg.sh
+```
+
+What that will do is setup a local tap for immuta and then "install" our
```
Comment (barryhammen) [inline depends/libhdfs3/osx/README.md]: Is the idea to contribute any of this back?  Probably don't want Immuta here (or anywhere) if that's the case

**Comment by KyleLilly** [comment]
Does it make sense to add the xattrs stuff to `depends/libhdfs3/src/client/hdfs.h` and `depends/libhdfs3/src/client/Hdfs.cpp` too?

File: depends/libhdfs3/osx/README.md (hunk 4 above)
Comment (sapanshah) [reply depends/libhdfs3/osx/README.md]: Yep, but the branding is how we get people to buy Immuta.

Done.

Comment (KyleLilly) [reply depends/libhdfs3/osx/README.md]: Should we remove the immuta stuff from the examples?

Comment (sapanshah) [reply depends/libhdfs3/osx/README.md]: Before I submit this back. I will probably remove the examples directory. This is just for you Kyle.

**Review by KyleLilly — APPROVED** [review]

**Pull request merged**